graphql/handler/apollotracing/tracer.go

```diff
@@ -60,28 +61,43 @@ func (a Tracer) InterceptField(ctx context.Context, next graphql.Resolver) (res
 		panic("missing tracing extension")
 	}
 
-	start := graphql.Now()
+	fc := graphql.GetFieldContext(ctx)
 
-	defer func() {
-		td.mu.Lock()
-		defer td.mu.Unlock()
-		fc := graphql.GetFieldContext(ctx)
+	if !fc.IsMethod && !isScalar(*fc.Field.Definition) &&
+		// Directives might do processing we want track
+		len(fc.Field.Directives) > 0 {
+		start := graphql.Now()
 
-		end := graphql.Now()
+		defer func() {
+			td.mu.Lock()
+			defer td.mu.Unlock()
 
-		td.Execution.Resolvers = append(td.Execution.Resolvers, ResolverExecution{
-			Path:        fc.Path(),
-			ParentType:  fc.Object,
-			FieldName:   fc.Field.Name,
-			ReturnType:  fc.Field.Definition.Type.String(),
-			StartOffset: start.Sub(rc.Stats.OperationStart),
-			Duration:    end.Sub(start),
-		})
-	}()
+			end := graphql.Now()
+
+			td.Execution.Resolvers = append(td.Execution.Resolvers, ResolverExecution{
+				Path:        fc.Path(),
+				ParentType:  fc.Object,
+				FieldName:   fc.Field.Name,
+				ReturnType:  fc.Field.Definition.Type.String(),
+				StartOffset: start.Sub(rc.Stats.OperationStart),
+				Duration:    end.Sub(start),
+			})
+		}()
+	}
 
 	return next(ctx)
 }
 
+func isScalar(definition ast.FieldDefinition) bool {
```

Comment:
Is there any easy to check if a field is a scalar?

Reply:
Not easily, you would need to store the `schema` passed into validate and lookup the named type from `graphql.GetFieldContext(ctx).Field.Definition.Type.Name()` in schema.Types.

But, you probably don't want to do that, scalars have no bearing on how hard something is to calculate. Take for example a `totalBalance` on an `Account` type that needs to go sum all the rows in the account.

OTOH there are Objects which are really cheap to calculate, like the User which can be pulled from context and costs nothing.